src/java.base/share/classes/java/lang/reflect/AccessFlag.java

```diff
@@ -0,0 +1,274 @@
+     * source modifier {@link Modifier#PRIVATE private}.
+     */
+    PRIVATE(Modifier.PRIVATE, true, Set.of(CONSTRUCTOR, METHOD, FIELD)),
+
+    /**
+     * The access flag {@code ACC_ABSTRACT}, corresponding to the
+     * source modifier {@code link Modifier#ABSTRACT abstract}.
+     */
+    ABSTRACT(Modifier.ABSTRACT, true, Set.of(TYPE, METHOD)),
+
+    /**
+     * The access flag {@code ACC_STATIC}, corresponding to the source
+     * modifier {@link Modifier#STATIC static}.
+     */
+    STATIC(Modifier.STATIC, true, Set.of(FIELD, METHOD)),
+
+    /**
+     * The access flag {@code ACC_FINAL}, corresponding to the source
+     * modifier {@link Modifier#FINAL final}.
+     */
+    FINAL(Modifier.FINAL, true, Set.of(FIELD, METHOD, PARAMETER, TYPE)),
+
+    /**
+     * The access flag {@code ACC_TRANSIENT}, corresponding to the
+     * source modifier {@link Modifier#TRANSIENT transient}.
+     */
+    TRANSIENT(Modifier.TRANSIENT, true, Set.of(FIELD)),
+
+     /**
+      * The access flag {@code ACC_VOLATILE}, corresponding to the
+      * source modifier {@link Modifier#VOLATILE volatile}.
+      */
+    VOLATILE(Modifier.VOLATILE, true, Set.of(FIELD)),
+
+    /**
+     * The access flag {@code ACC_SYNCHRONIZED}, corresponding to the
+     * source modifier {@link Modifier#SYNCHRONIZED synchronized}.
+     */
+    SYNCHRONIZED(Modifier.SYNCHRONIZED, true, Set.of(METHOD, CONSTRUCTOR)),
+
+    /**
+     * The access flag {@code ACC_NATIVE}, corresponding to the source
+     * modifier {@link Modifier#NATIVE native}.
+     */
+    NATIVE(Modifier.NATIVE, true, Set.of(METHOD)),
+
+    /**
+     * The access flag {@code ACC_INTERFACE}.
+     * @see Class#isInterface()
+     */
+    INTERFACE(Modifier.INTERFACE, false, Set.of(TYPE)),
+
+    /**
+     * The access flag {@code ACC_STRICT}, corresponding to the source
+     * modifier {@link Modifier#STRICT strictfp}.
+     */
+    STRICT(Modifier.STRICT, true, Set.of(METHOD, CONSTRUCTOR)),
+
+    /**
+     * The access flag {@code ACC_BRIDGE}
+     * @see Method#isBridge()
+     */
+    BRIDGE(0x00000040, false, Set.of(METHOD, CONSTRUCTOR)),
+
+    /**
+     * The access flag {@code ACC_VARARGS}.
+     * @see Executable#isVarArgs()
+     */
+    VARARGS(0x00000080, false, Set.of(METHOD, CONSTRUCTOR)),
+
+    /**
+     * The access flag {@code ACC_SYNTHETIC}.
+     * @see Class#isSynthetic()
+     * @see Executable#isSynthetic()
+     */
+    SYNTHETIC(0x00001000, false,
+              Set.of(TYPE, FIELD, METHOD, CONSTRUCTOR, ElementType.MODULE, PARAMETER)),
+
+    /**
+     * The access flag {@code ACC_MANDATED}.
+     */
+    MANDATED(0x00008000, false, Set.of(ElementType.MODULE, PARAMETER)),
+
+    /**
+     * The access flag {@code ACC_ANNOTATION}.
+     * @see Class#isAnnotation()
+     */
+    ANNOTATION(0x00002000, false, Set.of(TYPE)),
+
+   /**
+    * The access flag {@code ACC_ENUM}.
+    * @see Class#isEnum()
+    */
+    ENUM(0x00004000, false, Set.of(TYPE, FIELD)),
```

Comment:
These can use the **package‑private** constant fields from `java.lang.reflect.Modifier`: <https://github.com/openjdk/jdk/blob/ab623f8ae05c57b165d4d918d896dbdc5b4177e9/src/java.base/share/classes/java/lang/reflect/Modifier.java#L338-L347>

Reply:
Good suggestion; will be reflected in the next push.